### ISSUE TITLE
Issue with stream and field dropdowns

### ISSUE THREAD

**Issue by dylanmcreynolds** [Xi-CAM/Xi-cam.CatalogViewer#10]
Currently, the field names are only being pulled from the primary stream. So, if a stream has other fields, they won't show up. In fact, if the stream doesn't have at least one field that matches what's in the primary stream, then the stream does not show.

We should add something like this to collect all the fields:

    def get_all_fields(self, run_catalog):
        all_fields = {}
        for stream in list(run_catalog):
            all_fields.update(run_catalog[stream].metadata['descriptors'][0]['data_keys'])
        return all_fields

Once this is in, then we also need to track which fields came from which stream so that changing the selected stream shows you the appropriate field.

**Comment by dylanmcreynolds** [comment]
Since the sample method I wsrote above does not use any instance variables, it should probably be written as:

```
@staticmethod
def get_all_fields( run_catalog):
    all_fields = {}
    for stream in list(run_catalog):
        all_fields.update(run_catalog[stream].metadata['descriptors'][0]['data_keys'])
    return all_fields
```

**Comment by dylanmcreynolds** [comment]
Closed by #11
